### DISCUSSION TITLE
"FATAL: Tenant or user not found [SOLUTION]

### DISCUSSION THREAD

**Discussion by LucasKlemke** [supabase/supabase#30107] [General]
I was trying to create a new user to connect on supabase using sql queries, but I was getting the following error when trying to connect **"FATAL: Tenant or user not found** .

`CREATE USER joao WITH PASSWORD 'cachorroquente';`
`GRANT CONNECT ON DATABASE postgres TO joao;`
`GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO joao;`

❌ postgresql://`joao`:[password]@aws-0-sa-east-1.pooler.supabase.com:6543/postgres
🚨 **"FATAL: Tenant or user not found**

I realized that, in order to be able to connect with the created user, instead of inserting the user on the username field, I had to also insert the characters from the default user postgres, after the dot:

postgresql://`postgres.ooystigvanwktbjqnxhr`:[password]@aws-0-sa-east-1.pooler.supabase.com:6543/postgres

✅ postgresql://`joao.ooystigvanwktbjqnxhr`:[password]@aws-0-sa-east-1.pooler.supabase.com:6543/postgres

![Screenshot 2024-10-25 at 16 55 14](https://github.com/user-attachments/assets/1f22f429-a638-46b8-b854-3ca00d62df48)
![Screenshot 2024-10-25 at 16 55 37](https://github.com/user-attachments/assets/24ec2df0-5efc-4fe3-9d13-525a1d2ac372)

**Comment by putrafajarh** [comment]
Where to get username?

**Reply by BITRU** [reply]
to user you add (or postgres that exist)  DOT(.) and POOLER_TENANT_ID value from env.
so you have username:    postgres.POOLER_TENANT_ID

**Reply by nirmaaan** [reply]
Thank you. This saved me so much time.

**Reply by broncasrafa** [reply]
Thank you so much, worked for me.

**Comment by lanomkar** [comment]
I also got this error. I deleted the existing table in the database

**Comment by galessalazar** [comment]
I went in circles with this same error and what worked for me was changing the postgresql to just postgres.

**Comment by junowoz** [comment]
Hi everyone, I was facing this exact "FATAL: Tenant or user not found" error and was completely stuck for days. I wanted to share my solution as it adds a critical piece of information to the existing answers.

Like many here suggested, changing the username to the format `postgres.[project_ref]` was a necessary step. However, this alone did **not** solve the problem for me.

The final solution was discovering that the **Pooler Hostname is not always the default `aws-0...`**.

My specific project, created in the `us-east-2` region, was assigned a pooler hostname of:
`aws-1-us-east-2.pooler.supabase.com`

All my attempts were failing because I was using the common `aws-0...` address I saw in documentation and examples. The server was correctly reporting "Tenant not found" because my project existed on the `aws-1` server, not the `aws-0` server.

**THE SOLUTION:**
Go to **Project Settings > Database > Connection String** and copy the hostname **directly from your project's dashboard**. Do not assume it follows the standard pattern.

My final, working connection string used:
-   **User:** `postgres.[project_ref]`
-   **Host:** `aws-1-us-east-2.pooler.supabase.com` (Copied from my dashboard)
-   **Port:** `6543`

I hope this helps save someone else from this frustrating issue!

**Reply by RockStone** [reply]
Thanks for the tips. Have you tried linking the project through `supabase` cli? I keep getting the same error

```
╰─ supabase link        

Supabase CLI 2.50.2
Selected project: <project-ref>
Connecting to remote database...
failed to connect to postgres: failed to connect to `host=aws-1-us-east-2.pooler.supabase.com user=postgres.<project-ref> database=postgres`: server error (FATAL: Tenant or user not found (SQLSTATE XX000))
```

but not sure how to specify the connection string with the cli.

**Reply by arthurspk** [reply]
Bro, do you have an answer for this question? I have the same problem in my application.

**Reply by broncasrafa** [reply]
@arthurspk tenta a mesma solução do @LucasKlemke (junowoz). Funcionou para mim.

**Reply by arthurspk** [reply]
@broncasrafa cara, simplesmente esperei algumas horas e tentei novamente! o que pareceu para mim foi que o supabase estava dando um timeout, tentei com outros colegas de equipe e ocorreu a mesma coisa no ambiente deles, então esperei e tentei novamente.

**Reply by Akshola00** [reply]
thanks, this worked for me also, i changed from aws-0 to aws-1 and it initialized the connection @RockStone

**Reply by ssshanaka** [reply]
This actually was a life savor. I've been trying to connect with my database (on Supabase) for hours now. None of the AI agents or stackoverflow blogs has helped me so far, but this solution has worked. Thank you so much!

**Comment by SrDouglax** [comment]
My solution:
```bash
npx supabase gen types typescript --db-url "postgresql://postgres.[POOLER_TENANT_ID]:[POSTGRES_PASSWORD]@localhost:5432/postgres" > src/supabase/types.ts
```
Both POOLER_TENANT_ID and POSTGRES_PASSWORD are in .env of supabase self-hosted

Docs: https://supabase.com/docs/guides/self-hosting/docker#accessing-postgres

**Comment by mr-muddle** [comment]
same prob - Thanks bro

**Comment by mauro-n** [comment]
Thanks man! Tava ficando louco já kkk

**Comment by eeshankeni** [comment]
Btw (on self hosted) instead of port number it can also be the postgres pooler port number, I couldve sworn 5342 used to work but then there were a whole bunch of TLS issues. The pooler number is a 4 digit number that starts from 6 usually. Supabase db push with db-url works fine with 5432 but generate types does not
